### PR TITLE
[Bugfix] Fix Mamba model initialization and MLP Speculator weights loading

### PR DESCRIPTION
**Pull request by Isotr0py**
FIX https://github.com/vllm-project/vllm/pull/10327#issuecomment-2485999258 

Sorry for breaking the test :(

### CODE REVIEW

**Comment by Isotr0py** [comment]
@DarkLight1337 Seems that the failing Mamba loading is not a bug: https://buildkite.com/vllm/ci-aws/builds/11414#019342cb-092d-405f-8d84-89c9f345edd8/182-3271

I indeed can't find RMS norm in the model checkpoint: https://huggingface.co/tiiuae/falcon-mamba-tiny-dev/tree/main?show_file_info=model.safetensors, perhaps the tested mamba model was not initialized correctly.

File: vllm/model_executor/models/mamba.py [hunk 1]
```diff
@@ -48,7 +48,7 @@ def __init__(self,
                                 time_step_rank=config.time_step_rank,
                                 use_conv_bias=config.use_conv_bias,
                                 use_bias=config.use_bias,
-                                use_rms_norm=self.is_falcon_mamba,
```
Comment (Isotr0py) [inline vllm/model_executor/models/mamba.py:51]: I can't find RMS Norm from checkpoints in neither [tiiuae/falcon-mamba-7b](https://huggingface.co/tiiuae/falcon-mamba-7b) nor [tiiuae/falcon-mamba-tiny-dev](https://huggingface.co/tiiuae/falcon-mamba-tiny-dev/tree/main), so I set it to `False`.

Comment (tlrmchlsmth) [reply vllm/model_executor/models/mamba.py:51]: These should be set to true -- these norms are actually the only difference between falcon-mamba and mamba. 

We should probably handle this in the weight loader -- see discussion here: https://github.com/vllm-project/vllm/pull/9325#discussion_r1798293673

Comment (Isotr0py) [reply vllm/model_executor/models/mamba.py:51]: Thanks for the explanation! I will revert this and disable the weight loading tracking for mamba temporarily to have a quick fix on failing test.

Let's handle this in a separate PR.

**Review by tlrmchlsmth — APPROVED** [review]
Thanks for the fix!

**Pull request merged**